src/V3LinkJump.cpp

```diff
@@ -219,12 +221,21 @@ class LinkJumpVisitor final : public VNVisitor {
         killQueueCall->classOrPackagep(processClassp);
         AstStmtExpr* const killStmtp = new AstStmtExpr{fl, killQueueCall};
         nodep->addNextHere(killStmtp);
-        if (existsBlockAbove(targetName)) {
-            // process::kill doesn't kill the current process immediately, because it is in the
-            // running state. Since the current process has to be terminated immediately, we jump
-            // at the end of the fork that is being disabled
-            AstJumpBlock* const jumpBlockp = getJumpBlock(nodep->targetp(), false);
-            killStmtp->addNextHere(new AstJumpGo{fl, jumpBlockp});
+
+        // 'process::kill' does not immediately kill the current process
+        // executing the disable statement (because it's in the running state).
+        // If the disable statement is indeed executed by a process under the
+        // target AstFork, then jump to the end of that fork branch.
+        if (VN_IS(targetp, Fork)) {
```

Comment:
This seems a change.  "disable" of a block containing a fork should be disabling the contained fork, and it looks like this won't happen any more.

Reply:
I seem to have lost a comment on this (or I posted it somewhere else that I can't find), so repeating:

You are right, there was a new issue here, which sadly the tests were happy to accept, so I improved them.

FWIW the old code used to turn this fork with 2 branches:

```systemverilog
fork : a
   #1 disable a;
   $display("this should print");
join
```

into a fork with 1 branch:

```systemverilog
fork : a
   jumpblock
     #1 jumpgo;
     $display("this should print"); // Won't actually print
   endjumpblock
join
```

Anyhow, fixed that and improved the test. There is still one case that is broken both with the old and new versions. I will file an issue for that when this lands.